src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs

```diff
@@ -0,0 +1,332 @@
+        optionsProvider.TestGlobalOptions["build_property.RazorConfiguration"] = "Default";
+        optionsProvider.TestGlobalOptions["build_property.RootNamespace"] = "MyApp";
+        optionsProvider.TestGlobalOptions["build_property.RazorLangVersion"] = "Latest";
+        optionsProvider.TestGlobalOptions["build_property.GenerateRazorMetadataSourceChecksumAttributes"] = "false";
+
+        var additionalTexts = ImmutableArray<AdditionalText>.Empty;
+
+        foreach (var document in project.AdditionalDocuments)
+        {
+            var additionalText = new TestAdditionalText(document.Name, await document.GetTextAsync());
+            additionalTexts = additionalTexts.Add(additionalText);
+
+            var additionalTextOptions = new TestAnalyzerConfigOptions
+            {
+                ["build_metadata.AdditionalFiles.TargetPath"] = Convert.ToBase64String(Encoding.UTF8.GetBytes(additionalText.Path)),
+            };
+
+            optionsProvider.AdditionalTextOptions[additionalText.Path] = additionalTextOptions;
+        }
+
+        configureGlobalOptions?.Invoke(optionsProvider);
+
+        driver = driver
+            .AddAdditionalTexts(additionalTexts)
+            .WithUpdatedAnalyzerConfigOptions(optionsProvider);
+
+        return (driver, additionalTexts, optionsProvider);
+    }
+
+    protected static GeneratorRunResult RunGenerator(Compilation compilation, ref GeneratorDriver driver, params Action<Diagnostic>[] expectedDiagnostics)
+    {
+        driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out compilation, out _);
+
+        var actualDiagnostics = compilation.GetDiagnostics().Where(d => d.Severity != DiagnosticSeverity.Hidden);
+        Assert.Collection(actualDiagnostics, expectedDiagnostics);
+
+        var result = driver.GetRunResult();
+        return result.Results.Single();
+    }
+
+    protected static Project CreateTestProject(
+        Dictionary<string, string> additonalSources,
+        Dictionary<string, string>? sources = null)
+    {
+        var project = CreateBaseProject();
+
+        if (sources is not null)
+        {
+            foreach (var (name, source) in sources)
+            {
+                project = project.AddDocument(name, source).Project;
+            }
+        }
+
+        foreach (var (name, source) in additonalSources)
+        {
+            project = project.AddAdditionalDocument(name, source).Project;
+        }
+
+        return project;
+    }
+
+    private class AppLocalResolver : ICompilationAssemblyResolver
+    {
+        public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string>? assemblies)
+        {
+            foreach (var assembly in library.Assemblies)
+            {
+                var dll = Path.Combine(Directory.GetCurrentDirectory(), "refs", Path.GetFileName(assembly));
+                if (File.Exists(dll))
+                {
+                    assemblies!.Add(dll);
+                    return true;
```

Comment:
Why do we return after the first find vs. continuing? Is that just the nature of this interface contract?

Reply:
No, the contract seems to be to resolve all assemblies. But if I change that, the tests fail with:

<samp>
Pages_Index_razor.g.cs(12,74): error CS0433: The type 'ComponentBase' exists in both 'Microsoft.AspNetCore.Components, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' and 'Microsoft.AspNetCore.Razor.Test.ComponentShim.Compiler, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60'

Pages_Index_razor.g.cs(15,99): error CS0433: The type 'RenderTreeBuilder' exists in both 'Microsoft.AspNetCore.Components, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' and 'Microsoft.AspNetCore.Razor.Test.ComponentShim.Compiler, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60'
</samp>

Reply:
No, the contract seems to be to resolve all assemblies, I changed it.